src/solace_agent_mesh/gateway/http_sse/routers/background_tasks.py

```diff
@@ -0,0 +1,113 @@
+"""
+API Router for background task status queries.
+Supports background task status checking and active task listing.
+
+Note: The /tasks/{task_id}/events endpoint is defined in tasks.py
+to return the format expected by the workflow visualization frontend.
+"""
+
+import logging
+from fastapi import APIRouter, Depends, HTTPException, Query
+from sqlalchemy.orm import Session as DBSession
+
+from ....gateway.http_sse.dependencies import get_db
+from ....gateway.http_sse.repository.task_repository import TaskRepository
+from ....gateway.http_sse.repository.entities import Task
+from pydantic import BaseModel
+
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+class TaskStatusResponse(BaseModel):
+    """Response model for task status queries."""
+    task: Task
+    is_running: bool
+    is_background: bool
+    can_reconnect: bool
+
+
+@router.get("/tasks/{task_id}/status", response_model=TaskStatusResponse)
```

Comment:
Q: Is there a reason why this wasn't put in the tasks.py router with the other /tasks/{task_id}/... endpoints?